src/Controller/CompareProductController.php

```diff
@@ -45,7 +45,7 @@ public function comparePageContent(Request $request, SalesChannelContext $contex
     #[Route(path: '/compare/offcanvas', name: 'frontend.compare.offcanvas', options: ['seo' => false], defaults: ['_httpCache' => false, 'XmlHttpRequest' => true], methods: ['POST'])]
     public function offcanvas(Request $request, SalesChannelContext $context): Response
     {
-        $productIds = $request->get('productIds', []);
+        $productIds = json_decode($request->get('productIds', '[]'));
```

Comment:
```suggestion
        $productIds = json_decode($request->request->get('productIds', '[]'), true, 512, \JSON_THROW_ON_ERROR);
```

Why do we even need this change? Did the behaviour actually change?